content/docs/iac/build-with-pulumi/local-packages.md

```diff
@@ -0,0 +1,179 @@
+
+## What are Local Packages?
+
+Local packages are Pulumi packages with sdks generated on your computer, instead of being checked into a provider repository and published to the Pulumi Registry. All Pulumi packages include a [schema](/docs/iac/using-pulumi/pulumi-packages/schema/) that defines the resources and functions exposed by the package. These packages are frequently part of the Pulumi experience in several contexts:
+
+- When using parameterized providers like [pulumi-terraform-provider](https://www.pulumi.com/registry/packages/terraform-provider/).
+- When working with components as packages.
+- During package development and testing.
+- For private or organization-specific resources.
+
+Local packages allow you to generate SDKs in any Pulumi language for resources or components before they're published.
+
+## Adding Local Packages with `pulumi package add`
+
+The preferred way to add a local package is with the `pulumi package add` command. This command:
+
+1. Locally generates an SDK in your currently selected Pulumi language.
+1. Adds the package to your project configuration file (Pulumi.yaml).
+1. Prints instructions on how to link the package into your project.
+
+### Command Syntax
+
+```bash
+pulumi package add <provider|schema|path> [provider-parameter...] [flags]
+```
+
+### Adding a Package from Different Sources
+
+The `pulumi package add` command can accept the package source in multiple ways:
+
+#### From a Plugin Reference
+
+```bash
+pulumi package add PLUGIN[@VERSION]
+```
+
+This attempts to resolve a resource plugin, installing it on-demand similarly to:
+
+```bash
+pulumi plugin install resource PLUGIN [VERSION]
+```
+
+#### From a Local Path
+
+```bash
+pulumi package add ./my-provider
+```
+
+This executes the provider binary to extract its package schema.
+
+#### From a Schema File
+
+```bash
+pulumi package add ./my/schema.json
+```
+
+For details on the structure and syntax of Pulumi package schemas, refer to the [Schema Reference](/docs/iac/using-pulumi/pulumi-packages/schema/).
+
+#### From a Git Repository
+
+```bash
+pulumi package add example.org/org/repo.git/path[@version]
+```
+
+This clones the repo and executes the source. The version can be a tag (in semver format) or a Git commit hash.
+
+### Example: Using a Terraform Provider
+
+One of the most common uses of local packages is with the `terraform-provider` to access [any Terraform provider](https://www.pulumi.com/registry/packages/terraform-provider/):
+
+```bash
+pulumi package add terraform-provider hashicorp/random
+```
+
+This will generate a local SDK for Hashicorp's random provider, which you can then use in your project:
+
+```typescript
+import * as random from "random";
+
+new random.Pet("my-pet");
+```
+
+## Best Practices for Local Packages
+
+When working with local packages, consider these best practices:
+
+### Managing Generated SDKs
```

Comment:
This section is good, but could use an intro sentence here to ease the reader into the new topic.